### DISCUSSION TITLE
Passing server-sided data with custom server to page

### DISCUSSION THREAD

**Discussion by ionizer** [vercel/next.js#11491] [Help]
Hey there, sorry for asking a question that may have been asked multiple times already. But I am trying to make a sample project which uses [`oidc-provider` ](https://github.com/panva/node-oidc-provider) library. I'm all done with the OIDC configurations and only the frontend pages like login and authorization remaining to be developed, and I intend to use Next for this purpose. To follow their [example](https://github.com/panva/node-oidc-provider-example/blob/master/03-oidc-views-accounts/index.js), I need to use custom server to selectively render pages in a route. But problem is, now I could not find a way to pass the data from the express custom server which was obtained from `interactionDetails(req, res)` to my NextJS pages. I was thinking to use `query` parameter of Next's custom server `render` method to pass the data but then I've read about this usage being strongly discouraged and that it only supports string-(string | string[]) key-values with Typescript (I could probably `JSON.stringify` all the objects I want to pass in this case, but pretty sure this is basically hacking it). So, how can I pass the data properly from a custom server to my Next page?

I haven't really tried using `getServerSideProps` but it seems I can't even import my OIDC instance in my Next pages and will error out if I do back with `getInitialProps` before v9.3. But then again in order to follow the example, I need to selectively render the page (if user is logged in, show authorization page, otherwise show login page first) which I don't know how to achieve with pages alone.

**Comment by Wounded-Knee** [comment]
Same question.
